pkg/controller/jobs/pod/pod_webhook_test.go

```diff
@@ -0,0 +1,325 @@
+				Obj(),
+		},
+		"pod with owner managed by kueue (MPIJob)": {
+			initObjects:       []client.Object{defaultNamespace},
+			podSelector:       &metav1.LabelSelector{},
+			namespaceSelector: defaultNamespaceSelector,
+			pod: testingpod.MakePod("test-pod", defaultNamespace.Name).
+				Queue("test-queue").
+				OwnerReference(
+					"parent-mpi-job",
+					schema.GroupVersionKind{Group: "kubeflow.org", Version: "v2beta1", Kind: "MPIJob"},
+				).
+				Obj(),
+			want: testingpod.MakePod("test-pod", defaultNamespace.Name).
+				Queue("test-queue").
+				OwnerReference(
+					"parent-mpi-job",
+					schema.GroupVersionKind{Group: "kubeflow.org", Version: "v2beta1", Kind: "MPIJob"},
+				).
+				Obj(),
+		},
+		"pod with owner managed by kueue (PyTorchJob)": {
+			initObjects:       []client.Object{defaultNamespace},
+			podSelector:       &metav1.LabelSelector{},
+			namespaceSelector: defaultNamespaceSelector,
+			pod: testingpod.MakePod("test-pod", defaultNamespace.Name).
+				Queue("test-queue").
+				OwnerReference(
+					"parent-pytorch-job",
+					schema.GroupVersionKind{Group: "kubeflow.org", Version: "v1", Kind: "PyTorchJob"},
+				).
+				Obj(),
+			want: testingpod.MakePod("test-pod", defaultNamespace.Name).
+				Queue("test-queue").
+				OwnerReference(
+					"parent-pytorch-job",
+					schema.GroupVersionKind{Group: "kubeflow.org", Version: "v1", Kind: "PyTorchJob"},
+				).
+				Obj(),
+		},
+		"pod with owner managed by kueue (TensorFlowJob)": {
+			initObjects:       []client.Object{defaultNamespace},
+			podSelector:       &metav1.LabelSelector{},
+			namespaceSelector: defaultNamespaceSelector,
+			pod: testingpod.MakePod("test-pod", defaultNamespace.Name).
+				Queue("test-queue").
+				OwnerReference(
+					"parent-tf-job",
+					schema.GroupVersionKind{Group: "kubeflow.org", Version: "v1", Kind: "TFJob"},
+				).
+				Obj(),
+			want: testingpod.MakePod("test-pod", defaultNamespace.Name).
+				Queue("test-queue").
+				OwnerReference(
+					"parent-tf-job",
+					schema.GroupVersionKind{Group: "kubeflow.org", Version: "v1", Kind: "TFJob"},
+				).
+				Obj(),
+		},
+		"pod with owner managed by kueue (XGBoostJob)": {
+			initObjects:       []client.Object{defaultNamespace},
+			podSelector:       &metav1.LabelSelector{},
+			namespaceSelector: defaultNamespaceSelector,
+			pod: testingpod.MakePod("test-pod", defaultNamespace.Name).
+				Queue("test-queue").
+				OwnerReference(
+					"parent-xgboost-job",
+					schema.GroupVersionKind{Group: "kubeflow.org", Version: "v1", Kind: "XGBoostJob"},
+				).
+				Obj(),
+			want: testingpod.MakePod("test-pod", defaultNamespace.Name).
+				Queue("test-queue").
+				OwnerReference(
+					"parent-xgboost-job",
+					schema.GroupVersionKind{Group: "kubeflow.org", Version: "v1", Kind: "XGBoostJob"},
+				).
+				Obj(),
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			builder := utiltesting.NewClientBuilder(kubeflow.AddToScheme)
```

Comment:
```suggestion
			builder := utiltesting.NewClientBuilder()
```

I think it works well even if we don't load the kubeflow scheme to the client.